docs/DeveloperGuide.md

```diff
@@ -1,34 +1,833 @@
+<br>All available operations will be processed by the classes in the commands package. Every command class, like the `AddCommand` class, inherits from the `Command` class. Each individual command class is able to contain all the processing required for each command with the inputs passed in during the initiation of each command object. <br>
+<br>During runtime, the show related data is all stored in the `ShowList` class. The data is accessible and can be modified statically by all the command classes. The `ShowList` contains `Show` objects which describes the attributes of a show. 
+<br>Certain commands relating to the monitoring of the amount of time users spend watching shows depend on information from the `WatchTime` class. The class tracks the date and time remaining for the users to watch shows for the day. The time limit will be set by the user. <br>
+<img src = "images/storageDG/StorageClassDiagram.png" width = "700"> <br><br>
+This class diagram shows how the Storage class is setup as an example.<br>
+<br>On the initiation of WatchNext, the `Storage` object will be initiated and retrieves any user data that has been saved from previous runs. The data is stored in plain text and can be manually edited by advanced users. The data is stored in `data/showList.txt`. After the execution of every command, the `Storage` object calls upon the save command to automatically update the save data file. The commands relating to saving and loading data can be accessed from the `SaveState` interface.<br>
+<br>Throughout the lifespan of the program, various errors may occur. The `ErrorHandling` class stores the various errors that could occur. The expected errors usually stem from invalid user input or Input Output (IO) errors during file loading. The `Ui` class informs the users of the errors detected and suggests actions for rectification. <br>
+<img src = "images/designDG/designSequence.png" width = "700"> <br><br>
+This shows the flow of the program for one user command input.
+
+## 4. Implementation
+
+### AddCommand
+
+The `add` command allows users to add a new show which they are watching to the `ShowList`. It is invoked by the 
+`InputParser` class.
+
+Given below is an example of usage scenario of how the `add` command behaves at each step.
+
+**Step 1**
+
+* The user types in `add friends 2 10,10 30` , adding the `show` to the `Showlist` . The details added include the title
+of the show, the number of seasons of the show, the number of episodes in each season(separated by the comma) and 
+the duration of each episode. 
+
+The `InputParser` class calls the `parseInput` method to parse the command.
+
+> **[NOTE]** 
+>
+> `ArrayOutOfBoundsException` and `NullPointerException` is thrown when the number of arguments entered by the user is incorrect.
```

Comment:
I think it's nice that you have small snippets like this `**[NOTE]**` but I think you could include a section in your DG introducing the different notations used throughout the DG. :)